en/peripherals/mavlink_peripherals.md

```diff
@@ -1,24 +1,46 @@
-# MAVLink (OSD/Telemetry)
-
-## TELEM1 Port
-
-The telemetry 1 port defaults to 57600 baud, 8N1 and transmits a MAVLink
-stream.
-
-## TELEM2 Port
-
-The MAVLink settings default to OSD mode at 57600 baud. 
-
-The **drop-down menu** for the [SYS_COMPANION](../advanced_config/parameter_reference.md#SYS_COMPANION) parameter is used to set
-the optimal mode depending on the application scenario.
-
--   Companion computer mode at 921600 baud
--   Companion computer mode at 57600 baud
--   OSD mode at 57600 baud
--   Command / RC input mode (receive only) at 57600 baud
--   Normal telemetry mode at 57600 baud
-
-> **Note** OSD mode defines the MAVLink stream required by OSD (set of activated messages & their rates). The other modes are for other specific hardware (e.g. ESP8266 for WiFi), and use the optimal baud rate settings for their link.
-
-![QGC Telemetry Setup](../../images/qgc_telemetry_setup.png)
-
+# MAVLink Peripherals (GCS/OSD/Companion)
+
+Ground Control Stations (GCS), On-Screen Displays (OSD), Companion Computers, and other MAVLink peripherals interact with PX4 using separate MAVLink streams, sent via different serial ports.
+These communication channels are are configured using the [MAVLink parameters](../advanced_config/parameter_reference.md#mavlink).
+
+## MAVLink Instances
+
+In order to assign a particular peripheral to a serial port we use the (abstract) concept of a *MAVLink instance*. 
+Each instance can represent a particular set of streamed messages (see "mode" below); parameters are used to define the set of messages, the port used, data rate, etc.
+
+> **Note** At time of writing three instances are defined, which correspond to the 0, 1, 2 in the parameters listed below.
+
+The parameters for each instance are:
+- [MAV_X_CONFIG](../advanced_config/parameter_reference.md#MAV_0_CONFIG) - Set the serial port (UART) for this instance "X", where X is 0, 1, 2. 
+  It can be any unused port, e.g.: TELEM2, TELEM3, GPS2 etc. For more information see [Serial Port Configuration](../peripherals/serial_configuration.md).
+- [MAV_X_MODE](../advanced_config/parameter_reference.md#MAV_0_MODE) - Specify the telemetry mode/target (the set of messages to stream for the current instance and their rate).
+  The values are:

```

Comment:
>> What are they for?

> To configure which messages are streamed and at which rate. See https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_main.cpp#L1739.

No, I mean "specifically" - someone using this needs to be able to choose the right option. I have put some guesses in below but some of them aren't helpful. ie Normal and magic are empty, so why bother? Same with "Minimal" and "Config" - can you confirm WHEN you should set these?